x/lazyledgerapp/types/payformessage.go

```diff
@@ -138,6 +141,21 @@ func (msg *MsgWirePayForMessage) SignedTransactionDataPayForMessage(k uint64) (*
 	return &sTxMsg, nil
 }
 
+// parseMessage breaks the message into chunks and adds padding to msg.Message
+// if necessary
+func (msg *MsgWirePayForMessage) parseMessage() [][]byte {
+	// break message into shares
+	shares := chunkMessage(msg.Message)
+
+	// add padding if necessary
+	shares = addSharePadding(shares)
+
+	// modify the message to reflect padding
+	msg.Message = bytes.Join(shares, []byte{})
+
+	return shares
+}
```

Comment:
I would expect a method called `parseMessage` to take in bytes and output a message. This takes the `Message` field to return the shares but also modifies the `Message` field. The name should be different maybe.

What happens if it is called multiple times? Would it yield the same result (as essentially no padding is necessary)?

Reply:
How does extract-/inferShares sound? Is it essential that the `msg.Message` gets updated too? If not I'd suggest to remove that line.

Reply:
Or if the main purpose of the method is to pad `msg.Message` then call it `padMessage()` and remove the returned shares 🤔 The question then is if we could forget to call it?

Reply:
the message needs to be updated, and I agree, the name should change. I'm performing a small refactor that will allow us to name it `chunkMessage` as I think that describes it a bit better.